### PR TITLE
New mine variant for Magiclysm

### PR DESCRIPTION
**Pull request by Night-Pryanik**
#### Summary
Content "New mine variant"

#### Purpose of change
To add more content and homages.

#### Describe the solution
~Moria~ New mine consists of two levels (not counting the first underground level): 
- hallway with traces of battle, with skeletons of long-dead ~dwarves and orcs~ humanoids with their arms and armor in very bad shape.
- finale with long and winding pathway among the rocks, bottomless pits and streams of lava. Pathway concludes with a straight ~Bridge of Khazad-dûm~ bridge that leads to a closed room with a ~Balrog~ new `fiery nightmare` monster.
- The monster is very tough, very fast, has very high melee skills, an assortment of existing melee special attacks, and a new melee attack with its flaming whip. All attacks deal `HEAT` type of damage, so your armor most probably won't help you a lot. The new attack can even set you on fire (10% chance), and despite it's max duration is only 1 turn, it's almost certain death sentence for the unprepared.
- loot consists of new fabrication book which can learn you to forge a longsword from a ~mithril~ mysterious silvery metal. I shamelessly copied the stats for the material and the ingot from Magiclysm. I hope @KorGgenT won't yell at me for that.
#### Describe alternatives you've considered
Put it into Magiclysm instead of vanilla.

#### Testing
Debug-spawned new mine, teleported around it, checked for errors.

#### Additional context
Part of the hallway
изображение

Finale
изображение

### CODE REVIEW

File: data/json/items/book/fabrication.json [hunk 1]
```diff
@@ -547,5 +547,25 @@
     "max_level": 4,
     "intelligence": 9,
     "time": "30 m"
+  },
+  {
+    "id": "textbook_mithril",
+    "type": "BOOK",
+    "name": { "str_sp": "ᚲᚨᚲ:ᚲᛟᚹᚨᛏ:ᛗᛁᚠᚱᛁᛚ" },
+    "description": "This book is written on some runic language you have not a slightest idea about.  Fortunately, there are several very detailed colored pictures demonstrating the process of forging a sword from an ingot of silvery metal which leads you to an idea that this is some sort of textbook for blacksmiths.",
```
Comment (chaohedgehogs) [inline data/json/items/book/fabrication.json:555]:
```suggestion
    "description": "This book is written in a runic language you are unfamilliar with.  Fortunately, there are several very detailed pictures that demonstrate the process of forging a sword from an ingot of silvery metal.  You get the impression that this is some sort of textbook for blacksmiths.",
```

File: data/json/items/chemicals_and_resources.json [hunk 2]
```diff
@@ -1744,5 +1744,21 @@
     "price": 100,
     "price_postapoc": 5,
     "count": 100
+  },
+  {
+    "id": "mithril_ingot",
+    "type": "GENERIC",
+    "category": "spare_parts",
+    "name": "mithril ingot",
+    "description": "An ingot of mithril.  About 3 cm by 7 cm by 12 cm in size, ready to be used for various blacksmithing tasks.",
```
Comment (chaohedgehogs) [inline data/json/items/chemicals_and_resources.json:1753]:
```suggestion
    "description": "An ingot forged from an exotic silvery metal.  Aproximately 3 cm by 7 cm by 12 cm in size, ready to be used for various blacksmithing tasks.",
```

File: data/json/items/melee/swords_and_blades.json [hunk 3]
```diff
@@ -611,6 +611,16 @@
     "price": 13000,
     "price_postapoc": 50
   },
+  {
+    "id": "longsword_mithril",
+    "copy-from": "longsword",
+    "type": "TOOL",
+    "name": { "str": "longsword" },
+    "description": "This is a classic medieval sword, in between the lighter arming sword and later two-handed swords in size.  Most probably thanks to mysterious silvery metal it was forged from, it's very light yet harder than tempered steel.",
```
Comment (chaohedgehogs) [inline data/json/items/melee/swords_and_blades.json:619]:
```suggestion
    "description": "This is a classic medieval sword, the size being in-between the lighter arming sword and later two-handed swords.  Thanks to the mysterious silvery metal it was forged from, it is stronger than tempered steel while weighing substantially less.",
```

File: data/json/monsters/nether.json [hunk 4]
```diff
@@ -875,5 +875,56 @@
     "harvest": "exempt",
     "death_function": { "corpse_type": "NO_CORPSE", "message": "The %s melts away." },
     "flags": [ "HEARS", "GOODHEARING", "STUMBLES", "NOHEAD", "HARDTOSHOOT", "FLIES", "PLASTIC", "NO_BREATHE", "NOGIB" ]
+  },
+  {
+    "id": "mon_balrog",
+    "type": "MONSTER",
+    "name": { "str": "fiery nightmare" },
+    "description": "A colossal figure which seems to consist entirely of shadows engulfed by flames.  Its immense dark wings extend from wall to wall.  Two eyes, like two burning charcoals, are looking down on you with hatred.  In one hand it holds a huge sword, in the other it has a many-thonged whip, both glowing with red-hot flames.",
```
Comment (chaohedgehogs) [inline data/json/monsters/nether.json:883]:
```suggestion
    "description": "A colossal figure which seems to consist entirely of shadows engulfed in flames.  Its immense dark wings extend from wall to wall.  Two eyes that resemble burning coals are gazing down on you with hatred.  In one hand it holds a huge sword, in the other it has a many thonged whip, both glowing with red-hot flames.",
```

File: data/json/monsters/nether.json [hunk 5]
```diff
@@ -875,5 +875,56 @@
...
+    "aggression": 100,
+    "morale": 100,
+    "melee_skill": 10,
+    "melee_dice": 7,
+    "melee_dice_sides": 5,
+    "melee_damage": [ { "damage_type": "heat", "amount": 20 } ],
+    "melee_cut": 10,
+    "armor_bash": 12,
+    "armor_cut": 12,
+    "armor_bullet": 12,
```
Comment (Maleclypse) [inline data/json/monsters/nether.json:902]:
```suggestion
    "armor_bullet": 25,
```
I think animate shadows & flames should ignore more damage from small projectiles comparatively to what they take from cut and bash.

File: data/json/items/book/fabrication.json [hunk 6]
```diff
@@ -547,5 +547,25 @@
     "max_level": 4,
     "intelligence": 9,
     "time": "30 m"
+  },
+  {
+    "id": "textbook_mithril",
+    "type": "BOOK",
+    "name": { "str_sp": "ᚲᚨᚲ:ᚲᛟᚹᚨᛏ:ᛗᛁᚠᚱᛁᛚ" },
+    "description": "This book is written in a runic language that you are unfamilliar with.  Fortunately, there are several very detailed pictures that demonstrate the process of forging a sword from an ingot of silvery metal.  You get the impression that this is some sort of textbook for blacksmiths.",
```
Comment (actual-nh) [inline data/json/items/book/fabrication.json:555]:
```suggestion
    "description": "This book is written in a runic language unfamiliar to you.  Fortunately, there are several very detailed pictures that demonstrate the process of forging a sword from an ingot of silvery metal.  You get the impression that this is some sort of textbook for blacksmiths.",
```

File: data/json/monsters/nether.json [hunk 7]
```diff
@@ -875,5 +875,56 @@
...
+        "damage_max_instance": [ { "damage_type": "heat", "amount": 18, "armor_multiplier": 0.7 } ]
+      },
+      {
+        "id": "cut_throat",
+        "damage_max_instance": [ { "damage_type": "heat", "amount": 18, "armor_multiplier": 0.7 } ]
+      },
+      { "id": "flaming_whip" }
+    ],
+    "death_function": { "corpse_type": "NO_CORPSE", "message": "The %s melts away." },
+    "flags": [ "SEES", "WARM", "BASHES", "DESTROYS", "NO_BREATHE", "PUSH_MON", "PUSH_VEH", "FIREY", "PATH_AVOID_DANGER_2" ]
```
Comment (actual-nh) [inline data/json/monsters/nether.json:928]:
```suggestion
    "path_settings": { "avoid_traps": true, "avoid_sharp": true },
    "flags": [ "SEES", "WARM", "BASHES", "DESTROYS", "NO_BREATHE", "PUSH_MON", "PUSH_VEH", "FIREY", "PATH_AVOID_DANGER_1" ]
```
PATH_AVOID_DANGER_2 is unfortunately not working right currently. IIRC, see #48588 for more info

**Review by actual-nh — COMMENTED** [review]
I suspect this may better go into Magiclysm, but very nice in any event!

**Comment by SirPendrak** [comment]
Since this new sword is lighter, should it do a bit less bash damage?
And what about being able to use it with martial arts?

**Comment by Night-Pryanik** [comment]
Yeah, both suggestions are good. Will work on them a bit later today.

**Review by kevingranade — CHANGES_REQUESTED** [review]
> Put it into Magiclysm instead of vanilla.

Go with this please, too far into fantasy-land for vanilla.

**Pull request merged**